tools/generators/files_and_groups/src/Generator/Elements/Elements+RootElements.swift

```diff
@@ -0,0 +1,80 @@
+import PBXProj
+
+extension Generator {
+    enum Elements {}
+}
+
+extension Generator.Elements {
+    static func rootElements(
+        pathTree: PathTreeNode,
+        workspace: String,
+        elementAttributes: Generator.Environment.Elements.AttributesFunction
+    ) -> (
+        elements: [Element],
+        pathsToIdentifiers: [BazelPath: String],
+        knownRegions: Set<String>,
+        resolvedRepositories: [ResolvedRepository]
+    ) {
+        let resolvedRepositories: [ResolvedRepository] = [
+            .init(sourcePath: ".", mappedPath: workspace),
+        ]
+
+        // FIXME: Implement
+
+        var elements: [Element] = []
+        for node in pathTree.children {
+            switch node.name {
+            case "external":
+                elements.append(
+                    Generator.Elements.specialRootGroup(
+                        specialRootGroupType: .legacyBazelExternal,
+                        childIdentifiers: []
+                    )
+                )
+
+            case "bazel-out":
+                elements.append(
+                    Generator.Elements.specialRootGroup(
+                        specialRootGroupType: .bazelGenerated,
+                        childIdentifiers: []
+                    )
+                )
+
+            default:
+                break
+            }
+        }
+
+        // FIXME: Should this sorting happen here, or after the call?
+
+        // Elements are in the correct order, except for `.sortOrder`, so we
+        // need to sort on just that.
+        elements.sort { $0.sortOrder < $1.sortOrder }
+
+        return (
+            elements: elements,
+            pathsToIdentifiers: [:],
+            knownRegions: [],
+            resolvedRepositories: resolvedRepositories
+        )
+    }
+}
+
+struct Element: Equatable {
```

Comment:
Depending on the answer to my Q re `ElementsAttributes` this would change to `ElementsElements` which is interesting haha